temporalcli/devserver/freeport.go

```diff
@@ -44,7 +45,14 @@ func GetFreePort(host string) (int, error) {
 	// ports, making an extra connection just to reserve a port might actually
 	// be harmful (by hastening ephemeral port exhaustion).
 	if runtime.GOOS != "darwin" && runtime.GOOS != "windows" {
-		r, err := net.DialTCP("tcp", nil, l.Addr().(*net.TCPAddr))
+		// DialTCP(..., l.Addr()) might fail if machine has IPv6 support, but
+		// isn't fully configured (e.g. doesn't have a loopback interface bound
+		// to ::1). For safety, rebuild address form the original host instead.
+		tcpAddr, err := net.ResolveTCPAddr("tcp", fmt.Sprintf("%s:%d", host, port))
```

Comment:
My concern here is that we are not choosing a specific IP version based on the input string when binding, but we are when dialing. I am concerned about an ipv6-only machine (assuming they exist) if we blindly say `--ip 0.0.0.0` works for all IP versions. But it's not a real issue, we can just tell people to pass in `--ip ::` or something.